### PR TITLE
Add Basic Combat.

### PR DESCRIPTION
**Pull request by tclaar**
Give Chad health and a takeDamage method.
Make Snake invoke Chad's takeDamage method when attacking.
Make Stone Projectile invoke takeDamage method if applicable when hitting a target.

### CODE REVIEW

File: js/classes/entities/midground/weapon/projectile.js
Comment (NathanHinthorne) [inline js/classes/entities/midground/weapon/projectile.js:1]: Good idea with the second ACTION function

**Review by NathanHinthorne — APPROVED** [review]
Great changes! Thanks @tclaar

File: js/classes/entities/midground/chad.js [hunk 1]
```diff
@@ -87,6 +89,24 @@ class Chad {
...
+        return 100;
+    };
+
+    /** 
+     * Decrease the health of Chad by the provided amount and perform any necessary operations
+     * based on the new health value.
+     * 
+     * @param {number} amount the amount by which to decrease Chad's health
+     */
+    takeDamage(amount) {
```
Comment (calebkrauter) [inline js/classes/entities/midground/chad.js:103]: Looks good!

File: js/classes/entities/midground/enemies/village/snake.js [hunk 2]
```diff
@@ -132,7 +132,7 @@ class Snake {
                 this.action = "idle";
                 if (Date.now() - Snake.ATTACK_COOLDOWN * 1000 > this.lastAttack) {
```
Comment (calebkrauter) [inline js/classes/entities/midground/enemies/village/snake.js:133]: I like this, maybe we can make a constant in our project for 1000 as a conversion constant. Not necessary, just a thought.

File: js/classes/entities/midground/weapon/projectile.js [hunk 3]
```diff
@@ -86,8 +86,10 @@ class Projectile {
                 SIZE: new Vector(4, 4)
             },
             [Projectile.STONE]: {
-                ACTION: () => {
-                    console.log("stone");
+                ACTION: (target) => {
```
Comment (calebkrauter) [inline js/classes/entities/midground/weapon/projectile.js:89]: I like this! Very cool! Love how you can just change the Action for each ammo. @tclaar

File: js/classes/entities/midground/weapon/projectile.js [hunk 4]
```diff
@@ -166,7 +168,7 @@ class Projectile {
             GAME.entities.forEach((entity) => {
```
Comment (calebkrauter) [inline js/classes/entities/midground/weapon/projectile.js:168]: This is great! Could we make this its own function or put a comment saying why or what its doing?

File: js/classes/zone.js [hunk 5]
```diff
@@ -31,6 +31,9 @@ class Zone {
             GAME.addEntity(new Crosshair());
             GAME.addEntity(new Slingshot());
 
+            // temporary until the HUD is added
+            GAME.addEntity(new HealthBar(CHAD, Chad.MAX_HEALTH, Chad.SCALED_SIZE.x));
```
Comment (calebkrauter) [inline js/classes/zone.js:35]: Nice!

**Review by calebkrauter — APPROVED** [review]
Looks good! Just some minor suggestions for clarity but I like it!

**Pull request merged**